sabnzbd/postproc.py

```diff
@@ -948,47 +952,77 @@ def rar_renamer(nzo: NzbObject, workdir):
-            mychar = chr(ord(mychar) + 1)
-        logging.debug("Deobfuscate: rarsetname %s", rarsetname)
-
-        # Do the matching, layer by layer (read: rarvolnumber)
-        # So, all rar files with rarvolnr 1, find the contents (files inside the rar),
-        # and match with rarfiles with rarvolnr 2, and put them in the correct rarset.
-        # And so on, until the highest rarvolnr minus 1 matched against highest rarvolnr
-        for n in range(1, len(rarvolnr.keys())):
-            logging.debug("Deobfuscate: Finding matches between rar sets %s and %s" % (n, n + 1))
-            for base_obfuscated_filename in rarvolnr[n]:
-                matchcounter = 0
-                for next_obfuscated_filename in rarvolnr[n + 1]:
-                    # set() method with intersection (less strict): set(rarvolnr[n][base_obfuscated_filename]).intersection(set(rarvolnr[n+1][next_obfuscated_filename]))
-                    # check if the last filename inside the existing rar matches with the first filename in the following rar
-                    if rarvolnr[n][base_obfuscated_filename][-1] == rarvolnr[n + 1][next_obfuscated_filename][0]:
-                        try:
-                            rarsetname[next_obfuscated_filename] = rarsetname[base_obfuscated_filename]
-                            matchcounter += 1
-                        except KeyError:
-                            logging.warning(T("No matching earlier rar file for %s"), next_obfuscated_filename)
-                if matchcounter > 1:
-                    logging.info("Deobfuscate: more than one match, so risk on false positive matching.")
-
-        # Do the renaming:
-        for filename in rarsetname:
-            new_rar_name = "%s.%s" % (rarsetname[filename], volnrext[filename][1])
-            new_rar_name = os.path.join(workdir, new_rar_name)
-            new_rar_name = get_unique_filename(new_rar_name)
-            logging.debug("Deobfuscate: Renaming %s to %s" % (filename, new_rar_name))
-            renamer(filename, new_rar_name)
-            renamed_files += 1
+        return renamed_files
+
+    # numberofrarsets bigger than 1, so a mixed rar set, so we need pre-checking
+
+    # Sanity check of the rar set
+    highest_rar = sorted(rarvolnr.keys())[-1]
+    """
+    # First an easy check
+    if highest_rar != len(rarvolnr):
+        # there is a missing rar file, so
+        logging.warning("Incomplete rar set, so I can't deobfuscate")
+        return 0
+    """
+    # More intelligent check aka staircase check: number of rarsets should no go up, but stay the same or go down
+    how_many_previous = 1000  # 1000 rarset mixed ... should be enough ...
+    for rar_set_number in range(1, highest_rar + 1):
+        try:
+            how_many_here = len(rarvolnr[rar_set_number])
+        except:
+            # rarset does not exist at all
+            logging.warning("rarset %s is missing completely, so I can't deobfuscate.", rar_set_number)
```

Comment:
So this works well if one rar-set has 10 files and the other 15? Almost looks like it requires all sets to have same amount of rar's?

Reply:
Yes, that works. Please note that this was existing code (with just indentation changed)